Neos.Flow/Classes/Mvc/Routing/RoutingMiddleware.php

```diff
@@ -77,8 +65,7 @@ public function handle(ComponentContext $componentContext)
             $matchResults['@package'] = $this->packageManager->getCaseSensitivePackageKey($matchResults['@package']);
         }
 
-        $componentContext->setParameter(RoutingComponent::class, 'matchResults', $matchResults);
-        $httpRequest = $componentContext->getHttpRequest()->withAttribute(ServerRequestAttributes::ROUTING_RESULTS, $matchResults);
-        $componentContext->replaceHttpRequest($httpRequest);
+        $httpRequest = $request->withAttribute(ServerRequestAttributes::ROUTING_RESULTS, $matchResults);
+        return $next->handle($httpRequest);
```

Comment:
I think `$request = $request->with...` would be less error prone (i.e. don't introduce a new request instance but replace the other one). Or just:
```php
return $next->handle($request->withAttribute(ServerRequestAttributes::ROUTING_RESULTS, $matchResults));
```
?